### PR TITLE
Optimize the type of DialogOptions.backdrop

### PR DESCRIPTION
**Pull request by Heartade**
As mentioned in [this comment](https://github.com/ariakit/ariakit/issues/1698#issuecomment-1203456837), the previous type for DialogOptions.backdrop(`ElementType<ComponentPropsWithRef<"div">>`) is actually a subset of the type added in #1700. Replaced both with `ElementType<ComponentProps<"div">>`.

### CODE REVIEW

File: .changeset/dialog-backdrop-type-re.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"ariakit": minor
+---
+
+Updated the `backdrop` prop type on `Dialog` again to remove the original type which is a subset of the new type.
```
Comment (diegohaz) [inline .changeset/dialog-backdrop-type-re.md]: Can you delete this file and edit the original changeset to include this PR reference? The change hasn't been released yet, so we don't need two changelog entries. :)

Comment (Heartade) [reply .changeset/dialog-backdrop-type-re.md]: Commit pushed! New to this Changeset stuff, am I doing it right? 👀

**Review by diegohaz — APPROVED** [review]
Nice! Thanks!

File: .changeset/dialog-backdrop-type-re.md (hunk 1 above)
Comment (diegohaz) [reply .changeset/dialog-backdrop-type-re.md]: Yeah, that's right!

**Pull request merged**